Final Project/FinalProject/src/main/java/com/globant/app/ApiController.java

```diff
@@ -0,0 +1,316 @@
+	private ShoppingCartRepository cartRepository;
+	
+	@Autowired
+	private ProductRepository productRepository;
+	
+	/**
+	 * User that is currently log in the system
+	 */
+	private User userLogged;
+	
+	/**
+	 * Returns all users in the repository
+	 * @return
+	 */
+	@RequestMapping(value="/getUsers", method=RequestMethod.GET)
+	@ApiOperation(value="Finds all users", nickname="Get all users")
+	public @ResponseBody Collection<User> getUsers(){
+		return userRepository.findAll();
+	}
+	
+	/**
+	 * GET method for the user creation
+	 * @param model
+	 * @return
+	 */
+	@GetMapping("/addUser")
+	@ApiOperation(value="Add a new user", nickname="Add user")
+	public String registerForm(Model model){
+		model.addAttribute("User", new User());
+		return "addUser";
+	}
+	
+	/**
+	 * POST method for the user creation
+	 * @param user
+	 * @param model
+	 * @return
+	 */
+	@PostMapping("/addUser")
+	@ApiOperation(value="Add a new user", nickname="Add user")
+	public String register(@ModelAttribute User user, Model model){
+		User usernameInUse = userRepository.findByUsername(user.getUsername());
+		if (usernameInUse != null){
+			final String errorMessage = "Username already in use";
+			model.addAttribute("errorMessage", errorMessage);
+			model.addAttribute("User", new User());
+			return "addUser";
+		}
+		else{
+			User newUser = new User();
+			newUser.setId(user.getUsername());
+			newUser.setUsername(user.getUsername());
+			newUser.setPassword(user.getPassword());
+			userRepository.save(newUser);
+			return "redirect:/";
+		}
+	}
+	
+	/**
+	 * GET method for the login system
+	 * @param model
+	 * @return
+	 */
+	@GetMapping("/login")
+	@ApiOperation(value="Login a user to the system", nickname="Login")
+	public String login(Model model){
+		model.addAttribute("User", new User());
+		return "login";
+	}
+	
+	/**
+	 * POST method for the login system
+	 * @param user
+	 * @param model
+	 * @return
+	 */
+	@PostMapping("/login")
+	@ApiOperation(value="Login a user to the system", nickname="Login")
+	public String loginResult(@ModelAttribute User user, Model model){
+		User userExists = userRepository.findByUsername(user.getUsername());
+		if (userExists == null){
+			final String usernameError = "Username not found";
+			model.addAttribute("usernameError", usernameError);
+			model.addAttribute("passwordError", null);
+			model.addAttribute("User", new User());
+			return "login";
+		}
+		else{
+			if (userExists.getPassword().equals(user.getPassword())){
+				if (userExists.getCart() == null){
+					ShoppingCart cart = new ShoppingCart();
+					cart.setId(user.getUsername());
+					user.setCart(cart);
+					cartRepository.save(cart);
+					userRepository.delete(userExists);
+					userRepository.save(user);
+				}
+				model.addAttribute("loginUserCart", userExists.getCart());
+				model.addAttribute("loginUser", userExists);
+				userLogged = userExists;
+				return "userMenu";
+			}
+			else{
+				final String passwordError = "Invalid password";

```

Comment:
This should be constants at the beginning of the file or in another file with all the possible errors